src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,106 @@
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT id, title, price FROM books WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Book book = null;
+            if (resultSet.next()) {
+                book = getBook(resultSet);
+            }
+            return Optional.ofNullable(book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT id, title, price FROM books;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            List<Book> books = new ArrayList<>();
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                books.add(getBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of books "
+                    + "from books table.", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id= ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update a book: "
+                    + book, e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete a book by id " + id, e);
+        }
+    }
+
+    private Book getBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String title = resultSet.getString("title");
+        BigDecimal price = resultSet.getBigDecimal("price");
+        return new Book(id, title, price);
```

Comment:
To avoid code duplication when converting `ResultSet` to `Book`, it's good that you have moved this logic into a separate private method `getBook`. This adheres to the checklist item about avoiding code duplication.